src/diamond/handler/signalfx_handler.py

```diff
@@ -5,44 +5,57 @@
 
 #### Dependencies
 
- * urllib2
+ * signalfx (pip install signalfx)
 
 
 #### Configuration
 Enable this handler
 
- * handers = diamond.handler.httpHandler.SignalfxHandler
+ * handlers = diamond.handler.signalfx_handler.SignalfxHandler
 
  * auth_token = SIGNALFX_AUTH_TOKEN
+
  * batch_size = [optional | 300 ] will wait for this many requests before
      posting
-"""
 
+ * batch_max_interval = [optional | 10 ] Max interval (secs) to wait before
+     sending
+
+ * timeout = [optional | 1 ] Timeout for the HTTP Post calls.
+"""
 from Handler import Handler
-from diamond.util import get_diamond_version
-import json
+import collections
 import logging
 import time
-import urllib2
+
+try:
+    import signalfx
+except ImportError:
+    raise ImportError("Failed to load signalfx module. "
+                      "Install signalfx module - `pip install signalfx`")
+
+from diamond.util import get_diamond_version
 
 
 class SignalfxHandler(Handler):
 
     # Inititalize Handler with url and batch size
     def __init__(self, config=None):
-        Handler.__init__(self, config)
+        super(SignalfxHandler, self).__init__(config)
         self.metrics = []
+        auth_token = self.config['auth_token']
+        url = self.config['url']
         self.batch_size = int(self.config['batch'])
-        self.url = self.config['url']
-        self.auth_token = self.config['auth_token']
-        self.batch_max_interval = self.config['batch_max_interval']
+        self.batch_max_interval = float(self.config['batch_max_interval'])
+        self.timeout = self.config['timeout']
         self.resetBatchTimeout()
-        if self.auth_token == "":
```

Comment:
Don't we want an error still logged about no api key?  Or does the signalfx library handle that?